static/app/views/automations/components/actionFilters/subfiltersList.tsx

```diff
@@ -0,0 +1,192 @@
+          ...comparison,
+        };
+      }
+      return subfilter;
+    });
+    onUpdate({filters: newSubfilters});
+  }
+
+  return (
+    <Flex column>
+      <Flex column>
+        {subfilters.map((subfilter: Record<string, any>, i: number) => {
+          return (
+            <SubfilterRow
+              subfilter={subfilter}
+              subfilter_id={`${condition_id}.comparison.filters.${subfilter.id}`}
+              onRemove={() => removeSubfilter(subfilter.id)}
+              onUpdate={comparison => updateSubfilter(subfilter.id, comparison)}
+              key={subfilter.id}
+              isLastRow={i === subfilterCount - 1}
+            />
+          );
+        })}
+      </Flex>
+      <PurpleTextButton
+        borderless
+        icon={<IconAdd />}
+        size="xs"
+        onClick={() => addSubfilter()}
+      >
+        {t('Sub-filter')}
+      </PurpleTextButton>
+    </Flex>
+  );
+}
+
+interface SubfilterRowProps {
+  onRemove: () => void;
+  onUpdate: (comparison: Record<string, any>) => void;
+  subfilter: Record<string, any>;
+  subfilter_id: string;
+  isLastRow?: boolean;
+}
+
+function SubfilterRow({
+  subfilter,
+  subfilter_id,
+  onRemove,
+  onUpdate,
+  isLastRow,
+}: SubfilterRowProps) {
+  return (
+    <RowWrapper>
+      <Branch lastChild={isLastRow} />
+      <StyledRowLine>
+        <AutomationBuilderInputField
+          name={`${subfilter_id}.key`}
+          placeholder={t('key')}
+          value={subfilter.key}
+          onChange={(value: string) => {
+            onUpdate({
+              key: value,
+            });
+          }}
+        />
+        <AutomationBuilderSelectField
+          name={`${subfilter_id}.match`}
+          value={subfilter.match}
+          options={[
+            {
+              label: 'is',
+              value: MatchType.EQUAL,
+            },
+            {
+              label: 'is not',
+              value: MatchType.NOT_EQUAL,
+            },
+          ]}
+          onChange={(value: MatchType) => {
+            onUpdate({match: value});
+          }}
+        />
+        <AutomationBuilderInputField
+          name={`${subfilter_id}.value`}
+          placeholder={t('value')}
+          value={`${subfilter.value}`}
+          onChange={(value: string) => {
+            onUpdate({
+              value,
+            });
+          }}
+        />
+        {t('and')}
+        <DeleteButton
+          aria-label={t('Delete Subfilter')}
+          size="sm"
+          icon={<IconDelete />}
+          borderless
+          onClick={() => onRemove()}
+        />
+      </StyledRowLine>
+    </RowWrapper>
+  );
+}
+
+const RowWrapper = styled('div')`
+  display: flex;
+  align-items: center;
+  gap: ${space(1)};
+
+  :first-child {
+    margin-top: 3px;
+  }
+`;
+
+const StyledRowLine = styled(RowLine)`
+  padding: 3px 0;
+`;
+
+const DeleteButton = styled(Button)`
+  opacity: 0;
+
+  ${StyledRowLine}:hover &,
+  ${StyledRowLine}:focus-within &,
```

Comment:
Always select styles downwards not upwards. So in this case StyledRowLine should have a hover effect that makes the button inside it appear.